core/modules/wdb/QueryRunner.cc

```diff
@@ -322,29 +324,33 @@ void QueryRunner::_transmit(bool last, unsigned int rowCount, size_t tSize) {
     _result->set_attemptcount(_task->getAttemptCount());
 
     if (!_multiError.empty()) {
-        std::string chunkId = std::to_string(_task->msg->chunkid());
-        std::string msg = "Error(s) in result for chunk #" + chunkId + ": " + _multiError.toOneLineString();
+        string chunkId = to_string(_task->msg->chunkid());
+        string msg = "Error(s) in result for chunk #" + chunkId + ": " + _multiError.toOneLineString();
         _result->set_errormsg(msg);
         LOGS(_log, LOG_LVL_ERROR, msg);
     }
     _result->SerializeToString(&resultString);
-    _transmitHeader(resultString);
-    LOGS(_log, LOG_LVL_DEBUG, "_transmit last=" << last
-         << " resultString=" << util::prettyCharList(resultString, 5));
-
-    if (!_cancelled) {
-        // StreamBuffer::create invalidates resultString by using std::move()
-        xrdsvc::StreamBuffer::Ptr streamBuf(xrdsvc::StreamBuffer::createWithMove(resultString));
-        _sendBuf(streamBuf, last, transmitHisto, "body");
-    } else {
-        LOGS(_log, LOG_LVL_DEBUG, "_transmit cancelled");
+    {
+        // Nothing else can use this channel until this transmit is done.
+        lock_guard<mutex> streamLock(task->sendChannel->getStreamMutexRef());
+        _transmitHeader(resultString);
+        LOGS(_log, LOG_LVL_DEBUG, "_transmit last=" << last
+                << " resultString=" << util::prettyCharList(resultString, 5));
+
+        if (!_cancelled) {
+            // StreamBuffer::create invalidates resultString by using std::move()
+            xrdsvc::StreamBuffer::Ptr streamBuf(xrdsvc::StreamBuffer::createWithMove(resultString));
+            _sendBuf(streamBuf, last, transmitHisto, "body");
+        } else {
+            LOGS(_log, LOG_LVL_DEBUG, "_transmit cancelled");
```

Comment:
See my previous comment on suggested use of the predefined compiler-time constant `__func__` in favor of the actual function names.